### PR TITLE
Update object store to pull in fixes

### PR DESCRIPTION
**Pull request by austinzheng**
TODO:
- [x] Retarget `master` once #5395 is in.

### CODE REVIEW

**Comment by austinzheng** [comment]
Looks like a number of tests need to be updated in response to https://github.com/realm/realm-object-store/pull/573.

File: Realm/Tests/AsyncTests.mm [hunk 1]
```diff
@@ -721,11 +721,11 @@ - (void)testAddingBlockFromWithinNotificationBlock {
         }
     }];
 
-    // Triggers one call on each block
+    // Nested call is deferred.
     [self waitForNotification:RLMRealmDidChangeNotification realm:results.realm block:^{
         [self createObject:0];
     }];
-    XCTAssertEqual(calls, 2);
+    XCTAssertEqual(calls, 1);
```
Comment (tgoyne) [inline Realm/Tests/AsyncTests.mm]: I think `XCTAssertEqual(calls, 1); [results.realm refresh]; XCTAssertEqual(calls, 2);` will work to test this behavior.

Comment (austinzheng) [reply Realm/Tests/AsyncTests.mm]: Yes, that would better retain the original coverage of the test.

**Comment by austinzheng** [comment]
All failures are CI related.

**Pull request merged**